### PR TITLE
Support multi valued metrics

### PR DESCRIPTION
**Pull request by salvatore-campagna**
With this PR we handle collection of multi valued metrics
The idea is just to iterate on each value and collect all of
them as if each value belongs to a different document.

### CODE REVIEW

File: x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/RollupShardIndexer.java
Comment (csoulios) [inline x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/RollupShardIndexer.java:428]: I wonder if you could add one `instanceof` check outside the `for` loop would work.
Something like this: `if (values instanceof Number[])`

On the other hand we should totally remove the `instanceof` checks.

**Review by csoulios — COMMENTED** [review]
Change looks simple and clean.  And the code does the right thing when dealing with multi-values

I look forward to seeing how the code forms after leveraging the `MultiValueMode` enum.

**Comment by salvatore-campagna** [comment]
I need an approval for this. I didn't manage to use `MultiValueMode` because the change is larger than expected. As a result, I would proceed to merge this and work on other issues.

File: x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/RollupShardIndexer.java
Comment (nik9000) [reply x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/RollupShardIndexer.java:428]: Yeah, I think since we want to get the `instanceof` checks out entirely it's fine to do it either way.

**Review by nik9000 — APPROVED** [review]

**Comment by nik9000** [comment]
I *think* I'm ok to approve this while @csoulios is out for a bit. We have a bit of time before it fully lands in production and we expect a fair bit of refactoring over here eventually.

**Pull request merged**
